### PR TITLE
GH-40713: [Docs] Trim down ADBC page

### PR DESCRIPTION
**Pull request by lidavidm**
### Rationale for this change

Don't duplicate information across two docs sites.

### What changes are included in this PR?

Mostly replace the ADBC page with a link to the ADBC documentation.

### Are these changes tested?

N/A

### Are there any user-facing changes?

N/A
* GitHub Issue: #40713

### CODE REVIEW

**Comment by lidavidm** [comment]
I've incorporated some of the cut content here in https://github.com/apache/arrow-adbc/pull/2196

**Comment by vibhatha** [comment]
@lidavidm I am planning to review this PR. Could I get 2-3 days for this?

**Comment by ianmcook** [comment]
@github-actions crossbow submit preview-docs

**Review by WillAyd — APPROVED** [review]

File: docs/source/format/ADBC.rst [hunk 1]
```diff
@@ -117,185 +50,7 @@ backwards-incompatible versions such as 2.0.0, but which still
...
 Similarly, this documentation describes the ADBC API standard version
-1.0.0.  If/when an ABI-compatible revision is made
+1.1.0.  If/when an ABI-compatible revision is made
 (e.g. new standard options are defined), the next version would be
-1.1.0.  If incompatible changes are made (e.g. new API functions), the
+1.2.0.  If incompatible changes are made (e.g. new API functions), the
 next version would be 2.0.0.
-
-Related work
-============
```
Comment (ianmcook) [inline docs/source/format/ADBC.rst:126]: I hate to see all this great "Related work" content being deleted. Should we move it to a "Related work" page on the ADBC site? Some of this is already covered in the ADBC FAQ, but not all of it.

Comment (lidavidm) [reply docs/source/format/ADBC.rst:126]: I'll move it!

**Pull request merged**